src/mango/test/05-index-selection-test.py

```diff
@@ -108,41 +103,53 @@ def test_reject_use_index_ddoc_and_name_invalid_fields(self):
         selector = {
             "company": "Pharmex"
         }
-        try:
-            self.db.find(selector, use_index=[ddocid,name])
-        except Exception as e:
-            self.assertEqual(e.response.status_code, 400)
-        else:
-            raise AssertionError("did not reject bad use_index")
+        
+        resp = self.db.find(selector, use_index=[ddocid,name], return_raw=True)
+        self.assertEqual(resp["warning"], "{0}, {1} was not used because it is not a valid index for this query.".format(ddocid, name))
```

Comment:
in the test case above, you're checking that a value is added
```
# should still return a correct result
for d in r["docs"]:
    self.assertEqual(d["company"], "Pharmex")
```
We should be consistent?